File: Configuration/src/main/java/com/medecineWebApp/Configuration/payload/request/RegistrationRequest.java
package com.medecineWebApp.Configuration.payload.request;

import com.medecineWebApp.Configuration.models.role.Roles;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegistrationRequest {
    @NotEmpty(message = "FirstName is mandatory")
    @NotBlank(message = "FirstName is mandatory")
    private String firstname;
    @NotEmpty(message = "LastName is mandatory")
    @NotBlank(message = "LastName is mandatory")
    private String lastname;
    @Email(message = "Email is not formatted")
    @NotEmpty(message = "Email is mandatory")
    @NotBlank(message = "Email is mandatory")
    private String email;
    @NotEmpty(message = "Password is mandatory")
    @NotBlank(message = "Password is mandatory")

    private String password;

    private Set<Roles> roles ;

    private List<Long> departments = new ArrayList<>(); // ✅ Évite les erreurs si `null`;


}

File: Client/frontend/src/app/models/playload/registration-request.ts

export class RegistrationRequest {
    
    email: string;
    firstname: string;
    lastname: string;
    password: string;
    
  
    constructor(options: {
       
        email?: string;
        firstname?: string;
        lastname?: string;
        password?: string;
       
      } = {}) {
      
      this.email = options.email || '';
      this.firstname = options.firstname || '';
      this.lastname = options.lastname || '';
      this.password = options.password  || '';
     
    }
  }